### PR TITLE
22169 - api specs notice of withdrawal

### PR DESCRIPTION
**Pull request by eason-pan-bc**
*Issue #:* /bcgov/entity#22169

*Description of changes:*
**Adding items listed below to the api specs**
- Notice of Withdrawal Schema
- Sample request
- Sample responses

**Specs Render Screenshots:**
- Overview
image
- Sample Request
image
- Sample Request with Options
image
- Sample Response - successful
image
- Sample Response - 400
image
- Sample Response - 401
image
- Sample Response - 404
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
Shouldn't it be 404 if not found?

**Comment by severinbeauvais** [comment]
What about the optional data that could be posted / fetched? Like the 2 checkbox values and the court order.

**Comment by eason-pan-bc** [comment]
> Shouldn't it be 404 if not found?

Good point, we will need to adjust the validation code a bit to make it happen. It's kinda a straight forward change, should I change it and include it in this PR/ticket.

**Comment by eason-pan-bc** [comment]
> What about the optional data that could be posted / fetched? Like the 2 checkbox values and the court order.

Gonna add this example

**Comment by severinbeauvais** [comment]
> > Shouldn't it be 404 if not found?
> 
> Good point, we will need to adjust the validation code a bit to make it happen. It's kinda a straight forward change, should I change it and include it in this PR/ticket.

How about: update the spec now and update the code in one of the existing Legal API tickets?

**Comment by eason-pan-bc** [comment]
> > > Shouldn't it be 404 if not found?
> > 
> > 
> > Good point, we will need to adjust the validation code a bit to make it happen. It's kinda a straight forward change, should I change it and include it in this PR/ticket.
> 
> How about: update the spec now and update the code in one of the existing Legal API tickets?

Sounds good, added this task to [24989](https://app.zenhub.com/workspaces/entities---olga-65af15f59e89f5043c2911f7/issues/gh/bcgov/entity/24989)

**Comment by ArwenQin** [comment]
I would like to confirm the schema for the two checkbox values. Should they be placed within a sub-object or not?
"noticeOfWithdrawal": {
       "filingId": "153004",
       "courtOrder": {
           "fileNumber": "TT0010045",
           "effectOfOrder": "planOfArrangement"
       },
       "planOfArrangement": {
             "partOfPoa": true,
             "comeIntoEffect": true
   }
}

Or 
"noticeOfWithdrawal": {
       "filingId": "153004",
       "courtOrder": {
           "fileNumber": "TT0010045",
           "effectOfOrder": "planOfArrangement"
       },
         "partOfPoa": true,
         "comeIntoEffect": true
}

**Comment by eason-pan-bc** [comment]
> "noticeOfWithdrawal": { "filingId": "153004", "courtOrder": { "fileNumber": "TT0010045", "effectOfOrder": "planOfArrangement" }, "planOfArrangement": { "partOfPoa": true, "comeIntoEffect": true } }
> 
> Or "noticeOfWithdrawal": { "filingId": "153004", "courtOrder": { "fileNumber": "TT0010045", "effectOfOrder": "planOfArrangement" }, "partOfPoa": true, "comeIntoEffect": true }

I'm a bit confused, I only saw `courtOrder` as part of the NoW schema. https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/notice_of_withdrawal.json#L21C11-L21C122
Is there a NoW schema update? @severinbeauvais @ArwenQin

**Comment by ArwenQin** [comment]
Yes, the schema may need to be updated, as there are 2 new check boxes properties for NoW. 
For now, We are not sure whether we need to include the 2 properties, and whether in a sub-object.
Severin and I discussed, for now, we say Yes

**Comment by severinbeauvais** [comment]
> Is there a NoW schema update? @severinbeauvais @ArwenQin

Not yet. The courtOrder object is still there, and the 2 new boolean values may be optional or used for UI only. Still undecided but we don't need to add those to the schema yet.

**Comment by eason-pan-bc** [comment]
> Not yet. The courtOrder object is still there, and the 2 new boolean values may be optional or used for UI only. Still undecided but we don't need to add those to the schema yet.
> 
> Update: Yes we will use the 2 new properties. I prefer them in the root of the noticeOfWithdrawal object.

Since the api specs follows the schemas, how about I add the option of `courtOrder` for now. If we ended up updating the NoW schema, we can always come back to update the specs?

**Comment by severinbeauvais** [comment]
@ArwenQin Speaking of these 2 new properties, I think I was talking with Mihai and we were unsure whether we should save them or not.

On the one hand, staff should know that if either value is True then the NoW cannot be filed, so they won't even try.

On the other hand, if this filing ever becomes self-serve (ie, not staff-only) then we would use these values to prevent the NoW filing.

Please follow up with Mihai and Jacqueline on this. If staff don't need the properties then there's no need to save them. If regular users won't be allowed to file a NoW when either propery is True then, again, there's no need to save them.

**Comment by severinbeauvais** [comment]
> Since the api specs follow the schemas, how about I add the option of `courtOrder` for now. If we ended up updating the NoW schema, we can always come back to update the specs?

Sure. The courtOrder object is optional and we haven't decided about the 2 new properties yet. They may turn out to be for UI use only, so should not be part of the schema/spec.

**Comment by eason-pan-bc** [comment]
added the sample request with options, and updated the filing not found to 404

File: docs/business.yaml [hunk 1]
```diff
@@ -1498,6 +1503,27 @@ paths:
...
+                      foundingDate: '2023-07-12T17:31:58.000+00:00'
+                      identifier: BC1234567
+                      legalName: 1234567 B.C. LTD.
+                      legalType: BC
+                    noticeOfWithdrawal:
+                      filingId: 123456
+                      courtOrder:
+                        fileNumber: "A12345"
+                        effectOfOrder: planOfArrangement
+                        effectOfArrangement: ''
```
Comment (severinbeauvais) [inline docs/business.yaml:1526]: We aren't using `effectOfArrangement` and it isn't a court order property anyway. Remove it.

Ref: https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/court_order.json

Comment (eason-pan-bc) [reply docs/business.yaml]: updated

**Review by severinbeauvais — APPROVED** [review]

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**